src/js/components/Header/QueueList.react.js

```diff
@@ -108,42 +105,38 @@ export default class QueueList extends Component {
 
         const draggedTrack     = this.state.draggedTrack;
         const draggedOverTrack = this.state.draggedOverTrack;
+        const dragPosition     = this.state.dragPosition;
+
+        // If someone has a better idea...
+        const offset = (dragPosition === 'below' ? 1 : 0) + (draggedOverTrack < draggedTrack || (draggedOverTrack === draggedTrack && dragPosition === 'above') ? 1 : 0);
+
+        const newQueue = [...queue];
+
+        // remove draggedTrack
+        const movedTrack = newQueue.splice(queueCursor + 1 + draggedTrack, 1)[0];
```

Comment:
The whole queue is sent to `Queue`. So the correct moved track position in the queue is `queueCursor + 1 (for current track playing) + draggedTrack (dragged track index`)